module/apmawssdkgo/session.go

```diff
@@ -186,19 +271,84 @@ func send(req *request.Request) {
 	svc.setAdditional(span)
 
 	req.SetContext(ctx)
+	w.inflight.WithLabelValues(spanSubtype, req.Operation.Name).Inc()
 }
 
-func complete(req *request.Request) {
+func (w *awsWrapper) complete(req *request.Request) {
+	w.inflight.WithLabelValues(req.ClientInfo.ServiceName, req.Operation.Name).Dec()
+
 	ctx := req.Context()
 	span := apm.SpanFromContext(ctx)
 	if span.Dropped() {
 		return
 	}
 	defer span.End()
 
+	code := strconv.Itoa(req.HTTPResponse.StatusCode)
+	service := req.ClientInfo.ServiceName
+	operationName := req.Operation.Name
+
+	w.sentBytes.WithLabelValues(
+		service,
+		operationName,
+		code,
+	).Add(float64(requestSize(req)))
+
+	w.receivedBytes.WithLabelValues(
+		service,
+		operationName,
+		code,
+	).Add(float64(responseSize(req)))
+
 	span.Context.SetHTTPStatusCode(req.HTTPResponse.StatusCode)
 
 	if err := req.Error; err != nil {
 		apm.CaptureError(ctx, err).Send()
 	}
 }
+
+func requestSize(req *request.Request) int {
```

Comment:
this and `responseSize` were yanked from code in `client_golang`

Reply:
maybe move to `internal/apmhttputil`? also probably needs copyright attribution